android/app/src/main/java/org/fossasia/openevent/utils/AuthUtil.java

```diff
@@ -34,23 +36,45 @@
 
     private static Authenticator authenticator;
 
-    public static void signUpUser(Context context, String email, String password, ProgressBar progressBar) {
+    public static void signUpUser(String email, String password, MutableLiveData<Integer> signUpResponse) {
         APIClient.getOpenEventAPI().signUp(User.builder().email(email).password(password).build())
                 .subscribeOn(Schedulers.io())
                 .observeOn(AndroidSchedulers.mainThread())
                 .subscribe(
-                        user -> loginUser(context, email, password, progressBar),
+                        user -> {},
                         throwable -> {
-                            showProgressBar(progressBar, false);
                             showMessage(R.string.error_in_signing_up);
                             Timber.d(throwable.toString());
+                            signUpResponse.setValue(SignUpActivityViewModel.INVALID);
                         },
                         () -> {
-                            showProgressBar(progressBar, false);
                             showMessage(R.string.signed_up_successfully);
                             Timber.d("Signed up successfully");
+                            signUpResponse.setValue(SignUpActivityViewModel.VALID);
+                        });
```

Comment:
Since this observable emits a single, completed handler is same as onNext handler, move this up to `user -> {}`